File: 0_Project_CURRENT/Assets/scripts/WarpTubeScript_2.cs
using UnityEngine;
using System.Collections;

public class WarpTubeScript_2 : MonoBehaviour {
    public Material[] rens;
    public Color[] palette;
    public float[] uSpeeds;
    public float[] vSpeeds;

    void Start (){
	    //GameObject gameMaster = gameObject.Find("AA_gameMaster");
	    GameMaster gameMasterScript = GameObject.Find("AA_gameMaster").GetComponent<GameMaster>();
	    int world = gameMasterScript.worldNum;
	    for (int i=0; i< rens.Length ; i++) {
		    if(i==0) rens[i].SetColor ("_TintColor", palette[i+(world*10)]);
		    if(i==1) rens[i].SetColor ("_TintColor", palette[i+(world*10)]);
		    if(i==2) rens[i].SetColor ("_TintColor", palette[i+(world*10)]);
	    }
    }

    void Update (){
	    for (int i=0; i<rens.Length; i++) {
		    rens[i].mainTextureOffset = new Vector2 ((Time.time*-1*uSpeeds[i]) % 1, (Time.time*vSpeeds[i]) % 1);
	    }
    }
}

File: 0_Project_CURRENT/Assets/scripts/WarpTubeScript2.js
var rens: Material[];
var palette: Color[];
var uSpeeds: float[];
var vSpeeds: float[];

function Start(){
	var gameMaster: GameObject = gameObject.Find("AA_gameMaster");
	var gameMasterScript: GameMaster = gameMaster.GetComponent(GameMaster);
	var world : int = gameMasterScript.worldNum;
	for (var i: int=0; i<rens.length; i++) {
		if(i==0) rens[i].SetColor ("_TintColor", palette[i+(world*10)]);
		if(i==1) rens[i].SetColor ("_TintColor", palette[i+(world*10)]);
		if(i==2) rens[i].SetColor ("_TintColor", palette[i+(world*10)]);
	}
}

function Update () {
	for (var i: int=0; i<rens.length; i++) {
		rens[i].mainTextureOffset = Vector2 (Time.time*-1*uSpeeds[i], Time.time*vSpeeds[i]);
		rens[i].mainTextureOffset.x = rens[i].mainTextureOffset.x % 1;
		rens[i].mainTextureOffset.y = rens[i].mainTextureOffset.y % 1;
	}
}